__tests__/RemoveComponent.test.js

```diff
@@ -20,9 +20,47 @@ describe('Test <RemoveComponent/>', () => {
   afterAll(() => {
     onRemoveStub.restore();
   });
+  function mockItem(override) {
+    const props = Object.assign({}, minProps, override);
+    return <RemoveComponent { ...props } />;
+  }
 
   it('should render with base structure', () => {
     const wrapper = shallow(<RemoveComponent {...minProps} />);
     jestExpect(wrapper).toMatchSnapshot();
   });
+
+  it('should return empty span tag when readOnly set to true', () => {
+    const props = {
+      onRemove: stub(),
+      index: 0,
+      readOnly: true,
+      tag: {
+        id: 'Tags',
+        key: 'Tags',
+      },
+    };
+    const wrapper = shallow(<RemoveComponent { ...props } />);
+    expect(wrapper.html()).toBe('<span></span>');
+  });
+
+  it('should return Component when removeComponent is passed to RemoveComponent', () => {
+    const props = {
+      onRemove: stub(),
+      index: 0,
+      removeComponent: stub(),
+      tag: {
+        id: 'Tags',
+        key: 'Tags',
+      },
+    };
+    const wrapper = shallow(<RemoveComponent { ...props } />);
+    expect(wrapper.find('button')).toHaveLength(0);
```

Comment:
not clear what are you trying to test here ? Shouldn't this return a remove component instead ?

Reply:
What I'm trying to test is that there shouldn't be a `button` tag when `removeComponent` option is provided. If it doesn't make sense, I will remove this test. Thanks

Reply:
Instead of checking whether the button is rendered, pass a custom component (eg a span with some text) and check if that is rendered.

Reply:
Sorry for my lack of understanding, I think the test does not make sense to me. And the coverage is same without this test. Let me remove and push it again. Thanks